content/_partials/changelog-changes.html.haml

```diff
@@ -1,36 +1,42 @@
+:ruby
+  tags = {
+    'security' => 'Security',
+    'rfe' => 'Enhancement',
+    'majorrfe' => 'Major enhancement',
+    'bug' => 'Bug fix',
+    'majorbug' => 'Major bug fix'
+  }
+
 - if page.changes.empty?
   %em
     No notable changes in this release.
-- page.changes.each do | change |
-  %li{:class => change.type }
-    = change.message
-    - if change.references
-      - change.references.each_with_index do | reference, index |
-        - if index == 0
-          (
-        - else
-          ,&nbsp;
-        - if reference.issue
-          %a{:href => "https://issues.jenkins.io/browse/JENKINS-#{reference.issue}" }<>
-            = "issue #{reference.issue}"
-        - if reference.pull
-          %a{:href => "https://github.com/jenkinsci/jenkins/pull/#{reference.pull}" }<>
-            = "pull #{reference.pull}"
-        - else
-          %a{:href => reference.url }<>
-            - if reference.title
-              = reference.title
-            - else
-              = reference.url
-        - if index == change.references.count - 1
-          )
-    - elsif change.issue
-      (
-      %a{:href => "https://issues.jenkins.io/browse/JENKINS-#{change.issue}" }<>
-        = "issue #{change.issue}"
-      )
-    - elsif change.pull
-      (
-      %a{:href => "https://github.com/jenkinsci/jenkins/pull/#{change.pull}" }<>
-        = "pull #{change.pull}"
-      )
+- page.changes.group_by { |message| message["type"] }.each do | type, logs |
```

Comment:
I would recommend that at least the same grouping be done for the RSS/Atom changelog feed. I do not expect changelog authors to regularly look at that, so will likely miss if the experience there deteriorates (less grouping of entries in the data file).